src/server/bg_services/md_aggregator.js

```diff
@@ -30,13 +30,17 @@ async function run_md_aggregator(md_store, system_store, target_now, delay) {
     }
 
     let has_more = true;
+    let update_range = true;
+    let range = {};
 
     while (has_more) {
-        const range = await find_next_range({ target_now, system_store });
-        const update = range && await range_md_aggregator({ md_store, system_store, range });
-        if (update) {
+        if (update_range) range = await find_next_range({ target_now, system_store });
+        const changes = range && await range_md_aggregator({ md_store, system_store, range });
+        if (changes) {
+            const update = _.omit(changes, 'more_updates');
             await system_store.make_changes({ update });
-            await P.delay(delay);
+            if (!changes.more_updates) await P.delay(delay);
```

Comment:
Instead of 1 large bulk, you create multiple small bulks. This doesn't mean that we shouldn't wait.
In the case of 1000 buckets, you will create 10 simultaneous bulks which will take up mongo resources.
I think there should be a delay for these bulks as well.

Reply:
removed the if